neverbleed.h

```diff
@@ -53,6 +53,10 @@ int neverbleed_load_private_key_file(neverbleed_t *nb, SSL_CTX *ctx, const char
  * setuidgid (also changes the file permissions so that `user` can connect to the daemon, if change_socket_ownership is non-zero)
  */
 int neverbleed_setuidgid(neverbleed_t *nb, const char *user, int change_socket_ownership);
+/**
+ * set the cpu affinity for the neverbleed thread (returns 0 if successful)
+ */
+int neverbleed_setaffinity(neverbleed_t *nb, cpu_set_t *cpuset);
```

Comment:
`pthread_setaffinity_np` and `cpu_set_t` are only available on certain platforms.

In case of h2o, we use `#if defined(__linux__) || defined(__FreeBSD__) || defined(__NetBSD__)` as the guard.

I think it would be a good idea for us to have the same guard for the `neverbleed_setaffinity` function and related stuff.

Reply:
for linux, the man page specifies `To duplicate a CPU set, use memcpy(3).` which is why I took this approach in serializing. 

I didn't find anything in the man page for BSD, there is however there is `CPU_COPY`.

Now wondering if it's best to serialize to a common type, then send it over...

Reply:
Oh right, there's difference in the argument type as well.

Honestly I would not complain if our support for setaffinity getting reduced to linux only (in fear of having broken code on other platforms), but you've already read the code so I'm equally fine with what we have now.

Thank you for your efforts.